src/coreclr/jit/objectalloc.cpp

```diff
@@ -149,8 +154,18 @@ void ObjectAllocator::DoAnalysis()
 
     if (comp->lvaCount > 0)
     {
-        m_EscapingPointers         = BitVecOps::MakeEmpty(&m_bitVecTraits);
-        m_ConnGraphAdjacencyMatrix = new (comp->getAllocator(CMK_ObjectAllocator)) BitSetShortLongRep[comp->lvaCount];
+        m_EscapingPointers = BitVecOps::MakeEmpty(&m_bitVecTraits);
+        m_ConnGraphAdjacencyMatrix =
+            new (comp->getAllocator(CMK_ObjectAllocator)) BitSetShortLongRep[comp->lvaCount + m_maxPseudoLocals + 1];
+
+        // If we are doing conditional escape analysis, we also need to compute dominance.
+        //
+        if (CanHavePseudoLocals())
+        {
+            assert(comp->m_dfsTree != nullptr);
+            assert(comp->m_domTree == nullptr);
+            comp->m_domTree = FlowGraphDominatorTree::Build(comp->m_dfsTree);
```

Comment:
Is there any benefit to waiting until we actually want to use it to compute it? Or do we know at this point we are going to end up using it?

Reply:
Yeah we pretty much know we're going to use it.